ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/metric/BootstrapDumpMetricCollector.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.parse.repl.dump.metric;
+
+import org.apache.hadoop.hive.ql.parse.repl.metric.ReplicationMetricCollector;
+import org.apache.hadoop.hive.ql.parse.repl.metric.event.Metadata;
+
+/**
+ * BootstrapDumpMetricCollector.
+ * Bootstrap Dump Metric Collector
+ */
+public class BootstrapDumpMetricCollector extends ReplicationMetricCollector {
+  public BootstrapDumpMetricCollector(String dbName, String stagingDir, String policy,
+                                      long executionId,  long maxCacheSize) {
+    super(dbName, Metadata.ReplicationType.BOOTSTRAP, stagingDir, policy, executionId, 0, maxCacheSize);
```

Comment:
Why do we sub-classes BootstrapDumpMetricCollector and IncrementalDumpMetricCollector? Every thing is done at  super only. Can't we just set Metadata.ReplicationType appropriately?
Same for BootstrapLoadMetricCollector and IncrementalLoadMetricCollector as well.

Reply:
Both bootstrap and dumpexecution id is different. Kept it like this for extensibility in case in future we want to override something. Plus integration becomes simpler.